src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1655,28 +1753,28 @@ ErrorIfCoordinatorMetadataSetFalse(WorkerNode *workerNode, Datum value, char *fi
 
 /*
  * SetShouldHaveShards function sets the shouldhaveshards column of the
- * specified worker in pg_dist_node.
+ * specified worker in pg_dist_node. also propagates this to other metadata nodes.
  * It returns the new worker node after the modification.
  */
 static WorkerNode *
 SetShouldHaveShards(WorkerNode *workerNode, bool shouldHaveShards)
 {
-	return SetWorkerColumn(workerNode, Anum_pg_dist_node_shouldhaveshards,
-						   BoolGetDatum(shouldHaveShards));
+	return SetWorkerColumn(workerNode, Anum_pg_dist_node_shouldhaveshards, BoolGetDatum(
```

Comment:
Do we have any tests that uses `SetShouldHaveShards` and ensures that the MX workers also updates `shouldhaveshards` ? If not, we should add